### PR TITLE
[BC] Refactor `_KEEP_TEMPS` for reusability

### PR DESCRIPTION
**Pull request by tvmarino**
Patch that adds the option to keep temporary working_dir in env.py and adds working_dir as part of TimeStep in env.py

### CODE REVIEW

**Review by mtrofin — APPROVED** [review]
looks good, just 2 comments, feel free to land after addressing them.

**Comment by mtrofin** [comment]
Can you rename also the title of the change to e.g. "factor _KEEP_TEMP for reusability" or something in that vein

File: compiler_opt/rl/env_test.py [hunk 1]
```diff
@@ -161,6 +164,33 @@ def test_interactive_clang_session(self, mock_popen):
...
+    with env.clang_session(
+        _CLANG_PATH, _MOCK_MODULE, MockTask, interactive=True) as clang_session:
+      for _ in range(_NUM_STEPS):
+        obs = clang_session.get_observation()
+        working_dir = obs.working_dir
+        self.assertEqual(os.path.exists(working_dir), True)
+    self.assertEqual(os.path.exists(working_dir), False)
+
+    with flagsaver.flagsaver(
+        (env.compilation_runner._KEEP_TEMPS, '/tmp')):   # pylint: disable=protected-access
```
[the remark this answers is not in the document]
Comment (tvmarino) [reply compiler_opt/rl/env_test.py]: Can you check if this looks good now?

File: compiler_opt/rl/compilation_runner.py [hunk 2]
```diff
@@ -80,6 +80,18 @@ def __exit__(self, exc, value, tb):
     pass
 
 
+def get_workdir_context():
+  """Return a context which manages how the temperory directories are handled.
+
+  When the flag keep_temps is specified temporary directories are stored in
+  keep_temps."""
```
Comment (boomanaiden154) [inline compiler_opt/rl/compilation_runner.py:87]: Nit: The `"""` string terminator should be on a separate line.

**Review by boomanaiden154 — APPROVED** [review]
One minor nit, otherwise LGTM.

**Pull request merged**